### ISSUE TITLE
cpu memory leak problem when doing inference on large data.

### ISSUE THREAD

**Issue by aurooj** [CSAILVision/places365#65]
Hi, 

Nice work. I am trying to generate scene labels and attributes for video frames. The dataset is large-scale (~210K video frames). I am using your run_placesCNN_unified.py script. 

The only changes I made was to loop through the data folders and read video frame, and get model's predictions to save in a dictionary. I put model and input image on cuda to make it faster.

The code runs correctly except that it starts filling up the RAM very quickly and eats up all RAM eventually. I am using pytorch 0.4.0 with python 2.7. 
I have tried running with or without GPU, the behavior is same. 
Any pointers why it is happening?

**Comment by aurooj** [comment]
Here is the code snippet (print statements were commented out)
`   # load the test image
    root = 'path/to/video/folders'
    folders = os.listdir(root)

vcpt_dic = load_pickle('data/det_concepts.pickle')

for show in folders:
    clips = os.listdir(os.path.join(root, show))
    for clip in tqdm.tqdm(clips):
        clip_path = os.path.join(os.path.join(root, show, clip))
        frames = sorted(os.listdir(clip_path))
        id = 0
        for frame in frames:
            cur_scpt = ''
            img = Image.open(os.path.join(clip_path, frame))
            input_img = tf(img).unsqueeze(0)
            if input_img.size(1) < 3:
                b, ch, h, w = input_img.size()
                input_img = input_img.expand(b, 3, h, w)
            input_img = input_img.cuda()            

            # forward pass
            logit = model.forward(input_img)
            h_x = F.softmax(logit, 1).data.squeeze()
            probs, idx = h_x.sort(0, True)
            probs = probs.cpu().numpy()
            idx = idx.cpu().numpy()

            # print('RESULT ON ' + os.path.join(clip_path,frame))

            # output the IO prediction
            io_image = np.mean(labels_IO[idx[:10]])  # vote for the indoor or outdoor
            if io_image < 0.5:
                print('--TYPE OF ENVIRONMENT: indoor')                
                cur_scpt = cur_scpt + ',indoor'
            else:
                print('--TYPE OF ENVIRONMENT: outdoor')                
                cur_scpt = cur_scpt + ',outdoor'

            # output the prediction of scene category           
            cur_scpt = cur_scpt + ',' + classes[idx[0]]
            
            # output the scene attributes
            responses_attribute = W_attribute.dot(features_blobs[1])
            idx_a = np.argsort(responses_attribute)            
            scene_attr = [labels_attribute[idx_a[i]] for i in range(-1, -10, -1)]
            print(', '.join(scene_attr))            
            cur_scpt = cur_scpt + ',' + ','.join(scene_attr)            

            vcpt_dic[clip][id] = vcpt_dic[clip][id] + cur_scpt        
            id += 1          

    
        gc.collect()

with open('data/scene_lbls.pickle', 'wb') as f:
    pickle.dump(vcpt_dic, f)`

**Comment by aurooj** [comment]
This problem was solved when I clean features_blobs for each image in the loop.
